packages/cardano/src/cardano/api/parsers.test.ts

```diff
@@ -0,0 +1,299 @@
+  isNftMetadata,
+  parseFtMetadataRecord,
+  parseNftMetadataRecord,
+} from './parsers'
+
+describe('isMetadataFile', () => {
+  it('should return false if input is null', () => {
+    expect(isMetadataFile(null)).toBe(false)
+  })
+
+  it('should return false if input is undefined', () => {
+    expect(isMetadataFile(undefined)).toBe(false)
+  })
+
+  it('should return false if input is not an object', () => {
+    expect(isMetadataFile(42)).toBe(false)
+    expect(isMetadataFile('string')).toBe(false)
+    expect(isMetadataFile([])).toBe(false)
+    expect(isMetadataFile(true)).toBe(false)
+  })
+
+  it('should return false if name exists but is not a string', () => {
+    expect(
+      isMetadataFile({name: 123, mediaType: 'image/png', src: 'some-src'}),
+    ).toBe(false)
+  })
+
+  it('should return false if mediaType is not a string', () => {
+    expect(isMetadataFile({mediaType: 123, src: 'some-src'})).toBe(false)
+  })
+
+  it('should return false if src is not a string or array of strings', () => {
+    expect(isMetadataFile({mediaType: 'image/png', src: 123})).toBe(false)
+    expect(isMetadataFile({mediaType: 'image/png', src: [123]})).toBe(false)
+    expect(isMetadataFile({mediaType: 'image/png', src: ['valid', 123]})).toBe(
+      false,
+    )
+  })
+
+  it('should return true if only required fields are valid and others are not present', () => {
+    expect(isMetadataFile({mediaType: 'image/png', src: 'some-src'})).toBe(true)
+    expect(isMetadataFile({mediaType: 'image/png', src: ['some-src']})).toBe(
+      true,
+    )
+  })
+
+  it('should return true if all fields are valid', () => {
+    expect(
+      isMetadataFile({
+        name: 'some-name',
+        mediaType: 'image/png',
+        src: 'some-src',
+      }),
+    ).toBe(true)
+    expect(
+      isMetadataFile({
+        name: 'some-name',
+        mediaType: 'image/png',
+        src: ['some-src'],
+      }),
+    ).toBe(true)
+  })
+
+  it('should return true if name is not present but other required fields are valid', () => {
+    expect(isMetadataFile({mediaType: 'image/png', src: 'some-src'})).toBe(true)
+  })
+})
+
+describe('isFtMetadata', () => {
+  it('should return false if input is null', () => {
+    expect(isFtMetadata(null)).toBe(false)
+  })
+
+  it('should return false if input is undefined', () => {
+    expect(isFtMetadata(undefined)).toBe(false)
+  })
+
+  it('should return false if input is not an object', () => {
+    expect(isFtMetadata(42)).toBe(false)
+    expect(isFtMetadata('string')).toBe(false)
+    expect(isFtMetadata([])).toBe(false)
+    expect(isFtMetadata(true)).toBe(false)
+  })
+
+  it('should return false if name is not a string or is missing', () => {
+    expect(isFtMetadata({name: 123})).toBe(false)
+    expect(isFtMetadata({extra: 123})).toBe(false)
+  })
+
+  it('should return false if description exists but is not a string', () => {
+    expect(isFtMetadata({name: 'some-name', description: 123})).toBe(false)
```

Comment:
do you know if is it possible with `zod` to make it loose? but in that case I'd only use this feature if we can provide a fallback if that property didn't pass i.e `string` -> is `123` (invalid) defaulting to '' in the schema.

Reply:
k, gonna be moving this to product, right now it has the same behaviour as before, if it starts accepting invalid metadata it will be required some decisions around what to display `-` `invalid`, coerce into `string` i.e 123 -> `description: '123'` even though is wrong